src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContract.cs

```diff
@@ -169,155 +141,120 @@ internal static void ThrowInvalidDataContractException(string? message, Type? ty
-            get
-            { return _helper.KnownDataContracts; }
-
-            set
-            { _helper.KnownDataContracts = value; }
+            get => null;
         }
 
-        public virtual bool IsISerializable
+        internal GenericInfo? GenericInfo
         {
-            get { return _helper.IsISerializable; }
-
-            set { _helper.IsISerializable = value; }
+            [RequiresUnreferencedCode(DataContract.SerializerTrimmerWarning)]
+            get => _helper.GenericInfo;
+            set => _helper.GenericInfo = value;
         }
 
-        public XmlDictionaryString Name
+        public virtual DataContractDictionary? KnownDataContracts
         {
-            get
-            { return _name; }
-            set { _name = value; }
+            [RequiresUnreferencedCode(DataContract.SerializerTrimmerWarning)]
+            get => _helper.KnownDataContracts;
+            internal set => _helper.KnownDataContracts = value;
         }
 
-        public virtual XmlDictionaryString Namespace
+        public virtual bool IsISerializable
         {
-            get
-            { return _ns; }
-            set { _ns = value; }
+            get => _helper.IsISerializable;
+            internal set => _helper.IsISerializable = value;
         }
 
-        public virtual bool HasRoot
-        {
-            get
-            { return true; }
+        internal XmlDictionaryString Name => _name;
+
+        internal virtual XmlDictionaryString Namespace => _ns;
 
-            set
-            { }
+        internal virtual bool HasRoot
+        {
+            get => _helper.HasRoot;
+            set => _helper.HasRoot = value;
         }
 
         public virtual XmlDictionaryString? TopLevelElementName
         {
-            get
-            { return _helper.TopLevelElementName; }
-
-            set
-            { _helper.TopLevelElementName = value; }
+            get => _helper.TopLevelElementName;
+            internal set => _helper.TopLevelElementName = value;
         }
 
         public virtual XmlDictionaryString? TopLevelElementNamespace
         {
-            get
-            { return _helper.TopLevelElementNamespace; }
-
-            set
-            { _helper.TopLevelElementNamespace = value; }
+            get => _helper.TopLevelElementNamespace;
+            internal set => _helper.TopLevelElementNamespace = value;
         }
 
-        internal virtual bool CanContainReferences
+        internal virtual bool CanContainReferences => true;
+
+        internal virtual bool IsPrimitive => false;
+
+        public virtual bool IsDictionaryLike([NotNullWhen(true)] out string? keyName, [NotNullWhen(true)] out string? valueName, [NotNullWhen(true)] out string? itemName)
         {
-            get { return true; }
+            keyName = valueName = itemName = null;
+            return false;
         }
 
-        internal virtual bool IsPrimitive
+        public virtual List<DataMember>? Members
         {
```

Comment:
As there's no public setter, the only way to have a list here is to have the class create it itself. You have this member marked as nullable but if you returned an empty list instead, you could remove the nullable annotation. Would having an empty list mess up any logic elsewhere?

Reply:
This is a property that only existed on ClassDC and EnumDC before. If I were to go the empty-over-nullable route.... there is one place internally that would result in the execution of a property that we normally wouldn't execute. It looks innocuous, but you never know. I don't think I'd object to it.

But the bigger point here is that non-class/enum contracts don't need this at all. It seems silly to attach an empty list to them for no particular reason. The counter-argument there is that we could lazy-initialize it, since we never even try to access the list on non-class/enum contracts. The counter-counter-argument is... what if someone did access this property, and now they have an instance of a list. Would they expect that to be meaningful regardless of contract type?

I don't really like either path. Nullable vs empty. Such is the way when shoehorning API's. :/ Anyway, I'm leaving this alone for now, but could be persuaded if people feel strongly.